src/Reflectivity-Tools/MetaLinkIconStyler.class.st

```diff
@@ -0,0 +1,38 @@
+"
+I'm in charge to style an ast when there are metalink in the ast
+"
+Class {
+	#name : #MetaLinkIconStyler,
+	#superclass : #IconStyler,
+	#category : #'Reflectivity-Tools-Breakpoints'
+}
+
+{ #category : #defaults }
+MetaLinkIconStyler >> highlightColor [
+	^(Color orange alpha: 0.1)
+]
+
+{ #category : #defaults }
+MetaLinkIconStyler >> iconBlock: aNode [
+	^ [ aNode links inspect ]
+]
+
+{ #category : #defaults }
+MetaLinkIconStyler >> iconFor: aNode [
+	^ self iconNamed: #smallDebugIcon 
+]
+
+{ #category : #defaults }
+MetaLinkIconStyler >> iconLabel: aNode [
+	^ 'Metalinks...'
+]
+
+{ #category : #testing }
+MetaLinkIconStyler >> shouldStyleNode: aNode [
+	^ aNode hasLinks and: [ 
+		  aNode links anySatisfy: [ :link | 
+			  (link metaObject == Break or: [ 
+				   { 
+					   Watchpoint.
+					   ExecutionCounter } includes: link metaObject class ]) not ] ]
```

Comment:
it needs to be refactored.
If we will introduce kind of DebuggingTool hierarchy for Break, Watchpoint and others we will be able to remove parallel hierarchies for them. 
In this case we could have just one DebuggingIconStyler. It will check known tools installed on the node and it will style "unknown" metalinks in some default way like here.

Reply:
I agree.
I will add an issue after the PR is merged.